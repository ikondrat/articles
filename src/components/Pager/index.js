import React from 'react';
import Button from '@material-ui/core/Button';
import PropTypes from 'prop-types';
import {
    MoreHoriz
} from '@material-ui/icons';

const Pager = ({onClickMore, page, pages}) => (
    <div>
        {page < pages - 1 && 
            <Button color="primary" onClick={onClickMore}>
            Load more <MoreHoriz/>
        </Button>
        }
    </div>
)

Pager.propTypes = {
    onClickMore: PropTypes.func,
    page: PropTypes.number,
    pages: PropTypes.number
};

export default Pager;