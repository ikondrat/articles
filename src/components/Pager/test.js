import React from 'react';
import { shallow } from 'enzyme';
import Pager from '.';

it('renders correctly', () => {
  const wrapper = shallow(
      <Pager page={0} pages={10}/>
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders correctly on last page', () => {
  const wrapper = shallow(
      <Pager page={9} pages={10}/>
  );
  expect(wrapper).toMatchSnapshot();
});