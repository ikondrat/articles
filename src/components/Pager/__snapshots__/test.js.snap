// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pager
    page={0}
    pages={10}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Button)
        color="primary"
        onClick={undefined}
>
        Load more 
        <pure(MoreHorizIcon) />
</WithStyles(Button)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          "Load more ",
          <pure(MoreHorizIcon) />,
        ],
        "color": "primary",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        "Load more ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Button)
          color="primary"
          onClick={undefined}
>
          Load more 
          <pure(MoreHorizIcon) />
</WithStyles(Button)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Load more ",
            <pure(MoreHorizIcon) />,
          ],
          "color": "primary",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Array [
          "Load more ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`renders correctly on last page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pager
    page={9}
    pages={10}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": false,
    },
    "ref": null,
    "rendered": false,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": false,
      },
      "ref": null,
      "rendered": false,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
